### ISSUE TITLE
Step 1.b: in silico inference of gene regulatory networks

### ISSUE THREAD

**Issue by bastienchassagnol** [bastienchassagnol/DeCovarT#1]
8 de> Explanation: Why correlation can be misleading in GRNs. High expression correlation may arise because genes share a common upstream regulator, belong to the same cell-type programme, respond to the same environmental stimulus, or are co-regulated without direct interaction. Conversely, true regulatory edges may show weak correlation due to time lags, nonlinear effects, or context specificity.

> Categories of GRNs

<img width="672" height="851" alt="Image" src="https://github.com/user-attachments/assets/dd036209-7329-4f49-a056-be15c9a0fb10" />, from [Causal network analysis of omics data using prior knowledge databases](https://academic.oup.com/view-large/figure/544142076/bbaf654f1.tif)

<img width="1267" height="652" alt="Image" src="https://github.com/user-attachments/assets/42aa8092-6140-470d-95a9-07d05e14206f" />

# General review inferring GRNs

- [ ] [general review paper comparing approaches for inferring GRNs at the sample level](https://www.biorxiv.org/content/10.64898/2026.02.27.708608v1.full)
- [ ] [Gene regulatory networks: from correlative models to causal explanations](https://www.nature.com/articles/s41576-026-00939-1)

# General benchmarks

- [ ] [Reverse Engineering of Gene Regulatory Networks: A Comparative Study](https://pmc.ncbi.nlm.nih.gov/articles/PMC3171435/)
- [ ] emulation tests for tf links, https://academic.oup.com/nargab/article/8/2/lqag031/8550798?searchresult=1&login=false

# From single-cell to pseudo-bulk samples

- [ ] [Real-paired single-cell/bulk RNA-seq benchmark and a practical protocol for accurate cell-type deconvolution in human BAL samples](https://www.biorxiv.org/content/10.64898/2026.01.14.699304v1.full)
- [ ] Supplementary section of paper [Comprehensive Evaluation of Transcriptome-Based Cell-Type Quantification Methods for Immuno-Oncology](https://doi.org/10.1093/bioinformatics/btz363) for visual quality controls comparing single-cell with bulk profiles

- [ ] Chat GPT discussion, from Log Normal to Normal distribution, <https://chatgpt.com/c/69b58882-4ec8-8333-a21d-522926a9dbfa>

https://www.linkedin.com/safety/go/?url=https%3A%2F%2Fdoi%2Eorg%2F10%2E1101%2F2025%2E10%2E29%2E685322&urlhash=Knc0&mt=Yo9pQDGaq5SoGwKj7MyDBR-vGADPUAdENYB1kRMU05LqdQxXH-yNa9G0FmQJYuH51bzC6IPiCGQTU_JCMley-5bzSdQQNgcuN2YFspIAFEjKpmmjmxLMFdUUpvmDmr8A3nGQCIY_N1u0HlD2Y2u-m1fiDnR4vpsY&isSdui=true

# Static **gene regulatory networks**: 

## Seeds first, then path extension

- [ ] NeKo’s framework: 

    - Identify the minimal set of features to discriminate cell population( for instance, minimising the condition number, or global overlap)
    - Use them as seeds for the NeKo pipeline
    - Use automated Neko’s network reconstruction pipeline for retrieving back the edges of non connected vertices
    - Once the structure is known, infer the parameters with previous strategy. 

## Structure first, then infer the parameters

- [ ] [Scalable cell-specific coexpression networks for granular regulatory pattern discovery with NeighbourNet](https://genome.cshlp.org/content/36/4/785)

<img width="1025" height="1280" alt="Image" src="https://github.com/user-attachments/assets/001a886d-8c32-443b-b841-303d5e3d2e40" />

## Undirected, Markov models aka sparse Gaussian Markov Random Field

> [Introductory slides on the Graphical Lasso](https://bochang.me/gmrg/files/Bo_05152015.pdf)

> General theorem, from [GEOMETRY OF MAXIMUM LIKELIHOOD ESTIMATION IN
GAUSSIAN GRAPHICAL MODELS](https://projecteuclid.org/journals/annals-of-statistics/volume-40/issue-1/Geometry-of-maximum-likelihood-estimation-in-Gaussian-graphical-models/10.1214/11-AOS957.pdf)

  > In the Gaussian graphical model on G, the MLE of the covariance matrix  exists if and only if the G-partial sample covariance matrix SG can be completed to a positive definite matrix. Then the MLE ˆ is unique.  to that end, all submatrices corresponding to maximal cliques in MG must be positive definite, and G must be chordal. **If n ≥ q∗, the MLE exists with probability 1. If n<q, the MLE does not exist.**

> Review: [Contributions to Sparse Methods for Complex Data Analysis](https://hal.science/tel-01288976/), from Julien Chiquet. Key points: 

1.   Statistical issues raised by large feature spaces: **data scarcity**, aka the curse of high dimensionality, and **overfitting**, if the models are too complex, leading to poor generalisation
2. How to solve it? . Favor simple models, with interpratable structure, accounting for prior knowledge, performing dimension reduction, and efficient computation
3. Page 47: 3 methods to enforce sparsity in GGMs, constraint-based methods relying on independence tests, MCMC approaches (score-based), byt they do not scale, and finally l1 regularisation, that naturally accounts for the multuple testing issue as performed on the global precision matrix. 
4. Probablistic framework: Laplacian prior for edge distribution, and weights, see page 55 and 56 (EM implemention) + **variational EM approach**. 

### Inference

- [ ] [Gene Regulatory Networks from Multifactorial Perturbations Using Graphical Lasso: Application to the DREAM4 Challenge](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0014147)
  - applied to bulk transcriptomic data only

- [ ] [Maximum likelihood estimation of Gaussian graphical models: Numerical implementation and topology selection](https://www.seas.ucla.edu/~vandenbe/publications/covsel1.pdf): 
  - For chordal graphs, can be expressed as an unconstrained convex optimization problem, with a closed-form solution. 
  - Gradient-based approaches for **non-chordal graphs**
  - efficient implementation in high-dimensional, using chordal embeddings (convert your close to chardal graphs into truly chordal graphs)
  - searched property, as there's a factorisation including cliques for **Markov networks**

<img width="598" height="403" alt="Image" src="https://github.com/user-attachments/assets/1e43da32-6a6f-4a29-9e12-887edeb670d0" />

- [ ] [`SHINE` (Structure Learning for Hierarchical Networks)](https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1011118)

- [ ] [An empirical Bayes approach to inferring large-scale gene association networks](https://academic.oup.com/bioinformatics/article/21/6/754/199211)

**Cf poster of Blanche Francheterre from SMPG conference**

- Pair regular parsimonious inference of the precision matrix with deep learning in  [`GLAD`: Learning Sparse Graph Recovery](https://arxiv.org/pdf/1906.00271). Paper relies on a more general technique, further developed in [Algorithm Unrolling: Interpretable, Efficient Deep Learning for Signal and Image Processing](https://arxiv.org/pdf/1912.10557),  to render neural networks more interpretable.
- Inference using [poisson-LogNormal distributions](https://publications.ut-capitole.fr/id/eprint/26460/1/ImbertAlyssa2018.pdf)
- [Variational inference for sparse network reconstruction from count data](https://arxiv.org/pdf/1806.03120)
  - Introduce covariates, using offsets
  - Uses sparsity
- Same reference, competitors model:
  - Vanilla sparse GGM methods
  - Sparse log-linear graphical models, such as sparse LLM
  - `spieceasi` for compositional count data
- Projection to latent space
  - [VARIATIONAL INFERENCE FOR PROBABILISTIC POISSON PCA](https://arxiv.org/pdf/1703.06633)

- [ ] Standard graphical Lasso estimation, assuming sparse GRNs structure:[ https://link.springer.com/protocol/10.1007/978-1-4939-8882-2_6](https://link.springer.com/protocol/10.1007/978-1-4939-8882-2_6)

[R package](https://github.com/jchiquet/multivarNetwork) multivarNetwork (but not updated since 8 years)

    - Assumption that X (expression) follows a multivariate centered gaussian distribution
    - generalised the principle of GMM to multi-attribute (multi-modal inference of GRNs)  -> two nodes are connected if, and only if, all their attributes (in their framework, proteomic + transcriptomic) are both conditionally dependant, after accounting for the remaining interactions

- [ ] [`NExON-Bayes`: A Bayesian approach to network estimation informed by ordinal covariates](https://academic.oup.com/bioinformatics/advance-article/doi/10.1093/bioinformatics/btag345/8700505)

  -  applied to proteomics datasets
  - no latent clustering on the genes modules, but rather the addition of a covariate on the samples
  - sparsity induced on the off-diagonal through a continuous **spike-andslab** prior -> mixture, isntead pf the alternative option of a **point-mass**, direc distribution
  - covariate is ordinal, and playsa role on the inclusion of an edge through the **probit model**
  - use variational Bayesian EM, and blockwise descent gradient as a **surrogate** of the posterior distribution (equivzent to minimise the ELBO)
  - Hyperparameter selection through line search, and extend BIC minimisation

<img width="365" height="341" alt="Image" src="https://github.com/user-attachments/assets/6e77a234-7b96-401e-8b5d-6136897ae12d" />

- [ ] [Estimating heterogeneous gene regulatory networks from zero-inflated single-cell expression data](https://projecteuclid.org/journals/annals-of-applied-statistics/volume-16/issue-4/Estimating-heterogeneous-gene-regulatory-networks-from-zero-inflated-single-cell/10.1214/21-AOAS1582.full)

    - Bayesian mixture of GGM
    - Accounts for  cellular heterogeneity between samples
    - Deals with zero inflation distribution to relax the erroneous assumption of normality underlying transcriptomic distributions
    - deals with log normal distributions using Log Normal
    - model is identifiable, and inferred using Gibbs sampling
    - scRNASeq must be normalised for library sizes

### Add topological constraints

1. Hard constraints: fix the graph structure exactly, as mentioned in the literature, **covariance selection with known sparsity pattern**, an issue related to sparse Cholesky factorisation, solved through constrained MLE, cf [Sparse Linear Equations](https://cvxopt.org/userguide/spsolvers.html#sparse-linear-equations):

$$
\max_{\Theta \succ 0} ;;
\log \det \Theta - tr(S\Theta)
\quad \text{s.t. } \Theta_{ij} = 0 ;; \forall (i,j) \notin E
$$

2. **Hybrid strategy**, named edge-specific penalization, with weights modulating the strength of direct interaction. Weights = **penalty mask**:

$$
\max_{\Theta \succ 0} ;;
\log \det \Theta - \operatorname{tr}(S\Theta)
\sum_{i \neq j} \lambda_{ij} |\Theta_{ij}|
$$

3. Structured sparsity (tree structure, clusters, hubs, ...), with group graphical lasso, fused grapical lasso (across biological constraints)

> With l1 penalty, **wieghts are biased towards zero** (known penaly shrinkage issue)
> The structure must be **positive definite** (trees and chordal graphs always ok)

### Last step: from factorised to joint, global multivariate normal distribution

- [ ] With a DAG, and Bayesian model, rely on the [`bnlearn::gbn2mvnorm`](https://www.bnlearn.com/documentation/man/mvnorm.html)

## Latent-based framework (underlying TF structure)

- [ ] [PMF-GRN: a variational inference approach to single-cell gene regulatory network inference using probabilistic matrix factorization](https://link.springer.com/article/10.1186/s13059-024-03226-6)

<img width="1418" height="1410" alt="Image" src="https://github.com/user-attachments/assets/747fec65-4189-4c99-8c34-c42a5df6792d" />

### Review of prior knowledge

- [ ]  [Gene regulatory network inference in the era of single-cell multi-omics, from Julio Saez Rodriguez](https://www.nature.com/articles/s41576-023-00618-5)
- [ ]  [Causal network analysis of omics data using prior knowledge databases](https://academic.oup.com/bib/article/26/6/bbaf654/8371896)

### Biological contextualisation (from DAGs to causal networks)

<img width="1267" height="692" alt="Image" src="https://github.com/user-attachments/assets/68bc0c00-7456-4a1b-a9d5-e3e57d369e51" />

> To be tested in priority: NLBayes and TopoNPA, or NPA, since they use probabilistic frameworks. 

- [ ] [NPA: an R package for computing network perturbation amplitudes using gene expression data and two-layer networks](https://link.springer.com/article/10.1186/s12859-019-3016-x)
  - **Methodology seems unclear, with basic differential analysis. besides, only applied to bulk transcriptomics**

- [ ] <https://scispace.com/chat/07f6c266-050b-4728-9f69-bb8ad9689621>, and <https://scispace.com/search?q=What+are+the+advantages+and+limitations+of+using+the+graphical+lasso+technique+for+reconstructing+gene+regulatory+networks+from+single-cell+RNA+sequencing+data%3F>
- [ ] `greta`](https://github.com/saezlab/greta)
- [ ]  `SignalingProfiler 2.0`
- [ ] [Gene regulatory network integration with multi-omics data enhances survival predictions in cancer](https://academic.oup.com/bib/article/26/4/bbaf315/8188241), resort to `LIONESS` and `PANDA` algorithms

### Extensions: use a deep learning approach

- [ ] **Algorithm unrolling** 
  - https://www.biorxiv.org/content/10.1101/2020.04.23.058149v1
  - [GLAD: Learning Sparse Graph Recovery](https://arxiv.org/pdf/1906.00271)

- [ ] [Review of algorithm unrolling]
    -  Algorithm Unrolling: Interpretable, Efficient Deep Learning for Signal and Image Processing(https://arxiv.org/pdf/1912.10557)
    - https://hal.science/hal-04887464

### Model selection (report to section Model Selection)

- Stability Approach to Regularization Selection (StARS)
- eBIC 
- https://bastienchassagnol.github.io/CCCSTBench-Book/intro.html#compare-distributions

# Perspectives

## Directed, Causal, Bayesian-Network-based models

- [Joint estimation of causal effects from observational and intervention gene expression data](https://bmcsystbiol.biomedcentral.com/articles/10.1186/1752-0509-7-111)
- [Learning complex dependency structure of gene regulatory networks from high-dimensional microarray data with Gaussian Bayesian networks](https://www.nature.com/articles/s41598-022-21957-z)
- https://arxiv.org/pdf/2602.08629

<img width="685" height="607" alt="Image" src="https://github.com/user-attachments/assets/7b95d5eb-92da-4464-afba-0cb71066c3c7" />

- [Review causal knowledge](https://academic.oup.com/bib/article/26/6/bbaf654/8371896)
- Add perturbations: 
  - [A mini-review on perturbation modelling across single-cell omic modalities](https://www.sciencedirect.com/science/article/pii/S2001037024001417)
  - [`bnlearn`stucture](https://www.bnlearn.com/about/teaching/slides-bnshort.pdf)
  
<img width="791" height="551" alt="Image" src="https://github.com/user-attachments/assets/bdd457a9-5291-43be-b4ad-cbd50c2fa572" />

  - [Dissecting cellular ecosystem with single-cell CRISPR screens](https://pmc.ncbi.nlm.nih.gov/articles/PMC12680425/) -> main technologies sued belong to Perturb-seq, CROP-seq, ECCITE-seq, Direct-seq, [CRISP-Seq](https://www.sciencedirect.com/science/article/pii/S0092867416316117) and Mosaic-seq techniques + http://www.perturbase.cn/ stores single-cell perturbation datasets

- review of causal best practices, from [Best practices for moving from correlation to causation in ecological research](https://www.nature.com/articles/s41467-026-69878-z)

<img width="685" height="810" alt="Image" src="https://github.com/user-attachments/assets/2496f1a9-a03a-450a-9ffe-11db75614cac" />
 
# Perspectives for better capturing dynamics

## Real-time snapshots

> Explain why it's important: <https://www.linkedin.com/posts/%F0%9F%8E%AF-ming-tommy-tang-40650014_unlocking-the-true-potential-of-ai-and-activity-7448727853799497729-ixK1>

- [ ] [Snapshot in cell biology](https://www.cell.com/cell/pdf/S0092-8674(16)30208-2.pdf)

## Dynamic Gene Regulatory Networks

  -  [From time-series transcriptomics to gene regulatory networks: A review on inference methods](https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1011254), written by Vera and Malvina
  - [`Dictys`: dynamic gene regulatory network dissects developmental continuum with single-cell multiomics](https://www.nature.com/articles/s41592-023-01971-3)
  - [`TransMarker`: Unveiling dynamic network biomarkers in cancer progression through cross-state graph alignment and optimal transport](https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1013743)
  - [Gene Regulatory Network Inference Using Time-Stamped Cross-Sectional Single Cell Expression Data](https://www.sciencedirect.com/science/article/pii/S240589631632780X)
   - Uses Kolmogorov distances + Lasso regularisation
   - Compares its performance against Time Series Network Inference (TSNI), GEne Network Inference with Ensemble of trees (GENIE3), and JUMP3. 

## Dynamic, amd multi-omics!! 

- [ ] [Multi-omic network inference from time-series data](https://www.nature.com/articles/s41540-025-00591-1?fromPaywallRec=false#Fig1)

<img width="1997" height="810" alt="Image" src="https://github.com/user-attachments/assets/8b8c5ef2-eed0-4d38-ac7b-152acf79ed2c" />

## Combine Boolean with Bayesian inference

- [ ] [A Bayesian Noisy Logic Model for Inference of Transcription Factor Activity from Single Cell and Bulk Transcriptomic Data, `NLBayes`](https://www.biorxiv.org/content/10.1101/2023.05.03.539308v1.full#sec-12). **Major drawback**: not a true regulatory network, but rather a *regulon-level* analysis

<img width="1280" height="1040" alt="Image" src="https://github.com/user-attachments/assets/4e44b13a-3955-46b0-a30b-9bdbc7b1f543" />
